### DISCUSSION TITLE
How can a default value be set for a form.combobox tag?

### DISCUSSION THREAD

**Discussion by csalmeida** [josefarias/hotwire_combobox#167] [Q&A] [answered]
Hi 👋

I am currently trying to use `form.combobox` outsite of the context of a specific model and pass a default value to it.
What I mean by default is that for example on a `text_input` we can pass `:value` to it:

```ruby
<%= form.text_field :company, value: params[:company], class: "input" %>
```

Is it possible to do the same with the combobox like so?

```ruby
<%= form.combobox :location, Country.all, value: params[:location] %>
```
 
I noticed in the "A prefilled combobox" section in the documentation there's a way of assigning a pre-selected value by using a method? I didn't really understand that way but what I would like is to decide what the value would be myself if possible.
 
Usually the combobox was able to work this out because I was using the following pattern:

```ruby
<%= form_with model: model do |form| %>
```

However, now I am using a form with no model attached to it, it's purpose is to render inputs to filter a collection:

```ruby
<%= form_with url: search_employers_path, method: :get do |form| %>
```

I also think that the component might accept `:value` but it's not working in my case:
https://github.com/josefarias/hotwire_combobox/blob/main/app/presenters/hotwire_combobox/component.rb#L28

My goal is to pull the `:location` from the parameters, which is an ID for a row in a database that points to a `Country` and use that value from `params[:location]` to have that option selected on render.

I have attached a video below where we can see that when the form is submitted the selected value is cleared.

Is there a way to set the value or is there a better way of achieving this?
Any tips or feedback would be deeply appreciated. 🙏

https://github.com/josefarias/hotwire_combobox/assets/12223613/7f2af6b5-c093-4c0b-994f-d040a1b910ab

**Comment by csalmeida** [comment, marked as answer]
Okay, I think I was missing something on my side, after reading a bit more of the source code it did seem like `combobox` could take a `:value` parameter.

Then I found this in the documentation which made me think that maybe the `:value` needs to be an integer by default:

> ActiveRecord::Relation: the display is obtained by calling #to_combobox_display on each object and the value is obtained by calling #id. 

This change fixed it (`params[:location].to_i`):

```ruby
<%= form.combobox :location, Country.all, id: "location", label: nil, value: params[:location].to_i %>
```

Here's a demo where the value is being picked from `params[:location]` and it loads correctly:

https://github.com/josefarias/hotwire_combobox/assets/12223613/129eeb3a-1b92-4564-a14d-e502f183f40d

**Comment by josefarias** [comment]
Great digging! I'll reopen this so I remember to consider casting the value so you don't have to do that yourself.
